microsoft_authentication/auth/auth_utils.py

```diff
@@ -88,11 +88,14 @@ def validate_email(email):
     return "@" in email and email.split("@")[1] in settings.MICROSOFT["valid_email_domains"]
 
 
-def get_django_user(email, create_new=True, is_staff=True):
+def get_django_user(email, first_name, last_name, create_new=True, is_staff=True):
     if not validate_email(email=email):
         return
     try:
         user = User.objects.get(email=email)
+        user.first_name = first_name
+        user.last_name = last_name
+        user.save()
```

Comment:
```suggestion
        if not user.first_name and first_name:
                user.first_name = first_name
                user.save()
         if not user.last_name and last_name:
                 user.last_name = last_name
                 user.save()
```

No need to trigger user.save() every time.